frontend/src/containers/GroupApplication/Application.js

```diff
@@ -8,64 +8,53 @@ import {
 } from "src/components/styledFields";
 import readmeIfy from "src/components/ReadmeLogo";
 
-class Application extends Component {
-  componentDidMount() {
-    this.setState({
-      timeout: setInterval(() => {
-        sessionStorage.setItem(
-          "applicationText",
-          JSON.stringify({
-            ...JSON.parse(sessionStorage.getItem("applicationText")),
-            [this.props.group.name.toLowerCase()]: this.props.field.value,
-          })
-        );
-      }, 4000),
-    });
-  }
-  componentWillUnmount() {
-    clearInterval(this.state.timeout);
-  }
-  render() {
-    const {
-      responseLabel,
-      group,
-      field: { name, onChange, value },
-      form: { touched, errors, handleBlur },
-      disabled,
-    } = this.props;
-    const error = touched[name] && errors[name];
-    return (
-      <Container>
-        <LogoNameWrapper>
-          <Logo src={group.logo} />
-          <Name>{readmeIfy(group.name)}</Name>
-        </LogoNameWrapper>
-        {responseLabel && (
-          <ResponseLabel>{readmeIfy(responseLabel, true)}</ResponseLabel>
-        )}
-        <InputWrapper>
-          <FieldLabel htmlFor={group.name.toLowerCase()}>
-            Søknadstekst
-          </FieldLabel>
-          <InputArea
-            className="textarea"
-            type="textarea"
-            name={name}
-            id={name}
-            onChange={onChange}
-            onBlur={handleBlur}
-            placeholder="Skriv søknadstekst her..."
-            value={value}
-            error={error}
-            rows="10"
-            disabled={disabled}
-          />
-          <InputValidationFeedback error={error} />
-        </InputWrapper>
-      </Container>
+const Application = ({
+  responseLabel,
+  group,
+  field: { name, onChange, value },
+  form: { touched, errors, handleBlur },
+  disabled,
+}) => {
+  useEffect(() => {
+    sessionStorage.setItem(
+      "applicationText",
+      JSON.stringify({
+        ...JSON.parse(sessionStorage.getItem("applicationText")),
+        [group.name.toLowerCase()]: value,
+      })
```

Comment:
This is alot better, although I fear how this is going to work without any debouncing. It's probably fine, but it really should have a debounce.

An way to do it is to create a hook that stores the debounced value (untested):
```jsx

const useDebouncedState = (value, timeout = 500) => {
	const [debouncedState, setDebouncedState] = useState(value)

	useEffect(() => {
		const timer = setTimeout(() => setDebouncedState(value), timeout)
		return () => clearTimeout(timer);
	}, [value])
	
	return debouncedState;
}
```

Reply:
Did a little testing and that example seemed to work out pretty nice!